crates/noirc_frontend/src/hir_def/function.rs

```diff
@@ -125,6 +125,10 @@ pub struct FuncMeta {
     /// If this function is not in a contract, this is always 'Secret'.
     pub contract_function_type: Option<ContractFunctionType>,
 
+    /// This function's ability to be called from other contracts.
+    /// If this function is internal can only be called by itself.
+    pub is_contract_function_internal: Option<bool>,
```

Comment:
Lets rename this to `is_internal` since if it is internal it must be a contract function as well. We should also specify in the documentation that `None` implies the function is not a contract function. (looks like the docs for `contract_function_type` need to be updated too)